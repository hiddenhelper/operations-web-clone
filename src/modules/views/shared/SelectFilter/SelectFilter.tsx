import React, { memo, useState, useCallback } from 'react';

import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import MenuList from '@material-ui/core/MenuList';
import MenuItem from '@material-ui/core/MenuItem';

import ButtonFilter from '../ButtonFilter';
import { getDefaultValue } from '../../../../utils/generalUtils';

export interface ISelectFilterProps {
  optionList: { value: any; label: string }[];
  value: any;
  menuStyleClass?: string;
  onChange: (period: number) => void;
}

const SelectFilter = ({ optionList, menuStyleClass, value, onChange }: ISelectFilterProps) => {
  const [filter, setFilter] = useState(value);
  const onSelect = useCallback(
    option => {
      setFilter(option.label);
      onChange(option.value);
    },
    [onChange]
  );
  return (
    <ButtonFilter
      text={filter}
      transparentButton={true}
      render={({ isOpen, handleClose }) =>
        isOpen && (
          <ClickAwayListener disableReactTree={true} onClickAway={handleClose}>
            <MenuList className={getDefaultValue(menuStyleClass, '')} data-testid="select-filter-list" id="select-filter-list">
              {optionList.map((option, index) => (
                <MenuItem
                  disableRipple={true}
                  data-testid="select-filter-option"
                  onClick={event => {
                    onSelect(option);
                    handleClose();
                  }}
                  key={index}
                  value={option.value}
                >
                  {option.label}
                </MenuItem>
              ))}
            </MenuList>
          </ClickAwayListener>
        )
      }
    />
  );
};

export default memo(SelectFilter);
