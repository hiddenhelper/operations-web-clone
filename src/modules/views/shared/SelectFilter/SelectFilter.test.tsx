import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';

import SelectFilter, { ISelectFilterProps } from './SelectFilter';

describe('SelectFilter Component', () => {
  let props: ISelectFilterProps;

  beforeEach(() => {
    props = {
      optionList: [{ value: 'value', label: 'label' }],
      onChange: jest.fn(),
    };
  });

  it('should render', () => {
    const wrapper = render(<SelectFilter {...props} />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should change', () => {
    const wrapper = render(<SelectFilter {...props} />);

    act(() => {
      fireEvent.click(wrapper.getByTestId('button-filter-open'));
    });

    act(() => {
      fireEvent.click(wrapper.getAllByTestId('select-filter-option')[0]);
    });

    expect(props.onChange).toHaveBeenCalled();
  });
});
