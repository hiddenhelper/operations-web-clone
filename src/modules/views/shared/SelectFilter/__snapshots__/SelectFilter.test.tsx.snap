// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter Component should render 1`] = `
<div>
  <div
    class="makeStyles-floatingFilterWrapper-21 makeStyles-filterWrapper-20  floatingFilterWrapper"
  >
    <span
      class="makeStyles-floatingFilterButton-22"
      data-testid="controlled-button"
      tabindex="-1"
    >
      <button
        aria-controls="false"
        aria-haspopup="true"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-30"
        data-testid="button-filter-open"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium"
          >
            <svg
              class="makeStyles-dropdownIcon-24"
            >
              expand.svg
            </svg>
          </span>
        </span>
      </button>
    </span>
  </div>
</div>
`;
